projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/notification/EmailNotification.java

```diff
@@ -95,11 +70,47 @@ public void send(NotificationContent notificationContent) throws MessagingExcept
     }
   }
 
-  private boolean recipientsExist(Address[] recipients) {
-    if (recipients.length == 0) {
-      return false;
+  private void sendEmail(NotificationContent notificationContent, Address[] recipients)
+      throws MessagingException {
+    if (emailPropertiesConfiguration.isAuthEnabled()) {
+      sendAuthenticatedEmail(notificationContent, recipients);
+    } else {
+      sendUnauthenticatedEmail(notificationContent, recipients);
+    }
+  }
+
+  private void sendUnauthenticatedEmail(
+      NotificationContent notificationContent, Address[] recipients) throws MessagingException {
+    Transport transport = session.getTransport();
+    var mimeMessage = prepareMessage(notificationContent);
+    transport.send(mimeMessage, recipients);
+  }
+
+  private void sendAuthenticatedEmail(NotificationContent notificationContent, Address[] recipients)
+      throws MessagingException {
+    Transport transport = session.getTransport();
+    var mimeMessage = prepareMessage(notificationContent);
+    try {
+      transport.connect(
```

Comment:
do you need to connect every single time? 
You are always using the same session. 
So maybe you only need to connect the one time?

Reply:
Though this suggestion is certainly a possibility and I have considered it in general I think it's better to use and close the connection every time because SMTP servers will try to kill the session for various reasons - http://www.postfix.org/postconf.5.html#smtp_connection_reuse_time_limit , http://www.postfix.org/postconf.5.html#smtp_connection_cache_reuse_limit .To elaborate SMTP uses socket connection that remain open until explicitly closed - http://www.google.com/codesearch#ojJ6TKlKTCM/trunk/mail/src/java/com/sun/mail/smtp/SMTPTransport.java&type=cs . I also believe that in our case it's better to create a new connection instead of maintaining it open all the time, since we don't send that many emails anyway.

Reply:
makes sense. thanks